easybuild/easyconfigs/j/jupyterlab/jupyterlab-1.2.4-CrayGNU-19.10-cuda-10.1-batchspawner.eb

```diff
@@ -28,585 +22,198 @@ toolchain = {'name': 'CrayGNU', 'version': '19.10'}
-        }),
-    ('idna', '2.8', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/i/idna/'],
-        }),
-    ('chardet', '3.0.4', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/c/chardet/'],
-        }),
-    ('certifi', '2019.6.16', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/c/certifi/'],
-        }),
-    ('requests', '2.22.0', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/r/requests/'],
-        }),
-    ('jupyterhub', '1.0.0', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'use_pip': True,
-        'source_urls': ['https://pypi.python.org/packages/source/j/jupyterhub/'],
         }),
+    ('certipy', '0.1.3'),
+    ('entrypoints', '0.3'),
+    ('oauthlib', '3.0.2'),
+    ('pamela', '1.0.0'),
+    ('urllib3', '1.25.3'),
+    ('idna', '2.8'),
+    ('chardet', '3.0.4'),
+    ('certifi', '2019.6.16'),
+    ('requests', '2.22.0'),
+    ('jupyterhub', '1.0.0'),
 
 # widgets: note the lab extensions below
-    ('widgetsnbextension', '3.5.1', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/w/widgetsnbextension/'],
-        }),
-    ('ipywidgets', '7.5.1', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/i/ipywidgets/'],
-        }),
-# ipympl matplotlib widgets, requires lab extension 
-    ('ipympl', '0.3.3', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/i/ipympl/'],
-        }),
+    ('widgetsnbextension', '3.5.1'),
+    ('ipywidgets', '7.5.1'),
+
+# ipympl matplotlib widgets, requires lab extension
+    ('ipympl', '0.3.3'),
 
 # itkwidgets
-    ('param', '1.9.2', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/p/param/'],
-        }),
-    ('pyct', '0.4.6', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/p/pyct/'],
-        }),
-    ('colorcet', '2.0.2', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/c/colorcet/'],
-        }),
-    ('zstandard', '0.13.0', {
-        'req_py_majver': '3',
-        'req_py_minver': '6',
-        'source_urls': ['https://pypi.python.org/packages/source/z/zstandard/'],
-        }),
+    ('traittypes', '0.2.1'),
```

Comment:
Is this dependency needed earlier? Because it's already in the recipe later in the list.

Reply:
jupyterlab 1.1 already used 'configurable-http-proxy', '4.2.1' (together with 'jupyterlab-nvdashboard', '0.2.1')
perhaps @twrobinson can shed some light on this
Do we need the old versions?

Reply:
As this was never in production it is probably lagging in versioning. It should work with the configurable http proxy used in our 1.1.1 build. As for traittypes, it is a dep of itkwidgets and bqplot. Whichever occurs earlier in the eb file you can add it there. thanks!

Reply:
Thx Tim
traittypes was my bad, we don't need it twice. All fixed now